### DISCUSSION TITLE
Tracking of Source / Referrer

### DISCUSSION THREAD

**Discussion by misilot** [YOURLS/YOURLS#3529] [Q&A]
Hello,

Is there a way to add a query string to the end of the url track the referrer?

I'm hoping there is a way so we could add a specific `?tracking=digitalsignage`, `?tracking=tabletent`, `?tracking=flyer` or something similar? That way we could get statistics on what people are using to access the shortended url.

The sources could be different locations on our website, as well as direct links via a QR Code.

Thanks!

**Comment by dgw** [comment]
https://github.com/rinogo/yourls-keep-query-string to add tracking parameter(s) recognized by the long URL endpoint would be simplest.

Adding tracking like that inside YOURLS isn't something I could find an existing plugin for, unfortunately.
